src/main/java/com/zendesk/maxwell/Maxwell.java

```diff
@@ -76,12 +76,9 @@ private Position attemptMasterRecovery() throws Exception {
 				// load up the schema from the recovery position and chain it into the
 				// new server_id
 				MysqlSchemaStore oldServerSchemaStore = new MysqlSchemaStore(
-					context.getMaxwellConnectionPool(),
-					context.getReplicationConnectionPool(),
-					context.getSchemaConnectionPool(),
+					context,
```

Comment:
Please keep long argument lists rather than passing around `context` -- it makes the components far more reusable when using maxwell as a library.  I know it's a pain, but that's the reasoning.